### PR TITLE
[TRAFODION-1519]Use free tool to build windows ODBC

### PR DESCRIPTION
**Pull request by xwq**
1. Use Visual Studio Community 2013 instead of Visual Studio 2010.
Visual Studio Community 2013 is free.
2. Use open source tool inno setup to package windows ODBC Driver.
3. cleanup unused files

### CODE REVIEW

**Comment by hegdean** [comment]
My suggestion would be keep the set certificate directory code so we can provide an alternate location  for storing the client certificates rather than asking the customer to update the registry whenever ODBC driver installation is done.
My recommendation would be to create the DLL during the build step to c:\Build\winodbc64\...\lib folder so we can eliminate the  additional copy step in build_os.bat file

**Comment by xwq** [comment]
@hegdean  Now the inno setup script can't use the setCertificate any more. So this project will not be used any more. For the reason why inno setup script can't use setcertifate , you can ask Venkat for more information.  

By the way, You asked my delete the unused files about 3 weeks ago. Do you remember ? 

It doesn't make sense to change the target directory and it need change a lot of things . If you insist on it， you can create a JIRA  and commit a new pull request.

File: win-odbc64/odbcclient/README.txt [hunk 1]
```diff
@@ -17,68 +17,44 @@
...
-The final Win ODBC driver installer package can be found at %PACKDRIVER%\lib\x64\Release
+# @@@ END COPYRIGHT @@@
+
+Prerequisite:
+
+openssl >= 0.9.8
+zlib >= 1.2.8
+Microsoft Visual Studio >= 2013
+Inno Setup >= 5
+
```
Comment (arvind-narain) [inline win-odbc64/odbcclient/README.txt]: Please mention here the versions that were used to test.

File: win-odbc64/odbcclient/README.txt [hunk 2]
```diff
@@ -17,68 +17,44 @@
...
+Build steps
+
+1. In order to make the build script able to find these prerequisites,
+   some variables on the top of build_os.bat and pkg.bat need to be set accordingly.
```
Comment (arvind-narain) [inline win-odbc64/odbcclient/README.txt]: pkg.bat is removed - please update.

File: win-odbc64/odbcclient/README.txt [hunk 3]
```diff
@@ -17,68 +17,44 @@
...
+
+   Set variable INNO_SETUP_PATH to inno setup install folder, for example:
+	  set INNO_SETUP_PATH="C:\Program Files (x86)\Inno Setup 5"
+
+   Download vcredist_x64.exe from http://www.microsoft.com/en-us/download/details.aspx?id=40784 and copy it to C:\Build\winodbc64\redist
+
+2. To build, open a cmd window, change to win-odbc64\odbcclient and type
+	build_os.bat
+
+The final Win ODBC driver installer package can be found at %PACKDIR%
```
Comment (arvind-narain) [inline win-odbc64/odbcclient/README.txt]: Please test on a vanilla system to see if only the above steps are sufficient.

**Comment by arvind-narain** [comment]
@xwq Suggestion by @hegdean are valid -

1.  It will be better to keep set certificate directory code in place so that customer can specify where to store the certificate. Inno setup script didn't handle this earlier but this needs to be put in place. 

2. C:\Build\winodbc64 should not be hard coded in our code/installation steps. There is %PACKDIR% defined that should be used.  Installer.iss should take in a parameter so that the hard coded location can be removed. Same should be used in build_os.bat - we currently use only in one step. Do also please update the comments in README.txt to match the definition of PACKDIR - currently it says to point to the source.

I know you have made it easy for the user to install the windows driver - really appreciate  your contribution - but we also need to make sure that the instructions are clear and works for all sites - hence should avoid hard coded locations. Any steps we need to make the installation work should be clearly specified in the README.txt

**Comment by rkakarlamudi** [comment]
+1



*From:* arvind-narain [mailto:notifications@github.com]
*Sent:* Friday, October 16, 2015 11:24 AM
*To:* apache/incubator-trafodion <incubator-trafodion@noreply.github.com>
*Subject:* Re: [incubator-trafodion] [TRAFODION-1519]Use free tool to build
windows ODBC (#126)



@xwq <https://github.com/xwq> Suggestion by @hegdean
<https://github.com/hegdean> are valid -

1.      It will be better to keep set certificate directory code in place
so that customer can specify where to store the certificate. Inno setup
script didn't handle this earlier but this needs to be put in place.

2.      C:\Build\winodbc64 should not be hard coded in our
code/installation steps. There is %PACKDIR% defined that should be used.
Installer.iss should take in a parameter so that the hard coded location
can be removed. Same should be used in build_os.bat - we currently use only
in one step. Do also please update the comments in README.txt to match the
definition of PACKDIR - currently it says to point to the source.

I know you have made it easy for the user to install the windows driver -
really appreciate your contribution - but we also need to make sure that
the instructions are clear and works for all sites - hence should avoid
hard coded locations. Any steps we need to make the installation work
should be clearly specified in the README.txt

—
Reply to this email directly or view it on GitHub
<https://github.com/apache/incubator-trafodion/pull/126#issuecomment-148792822>
.

**Comment by arvind-narain** [comment]
Thanks for the update. With the version change another place to touch would be the exe file TFODBC64-1.2.0.exe.

@hegdean - could you please go through the recent change and comment ? If after the version change you think everything is ok @sureshsubbiah could help in merging ? I have been offline for some time and will be for next couple of days.

**Comment by arvind-narain** [comment]
Thanks for handling the version change and catching the missing copyright.

**Comment by hegdean** [comment]
Changes look good to me. 

Some minor changes which can be addressed in the next release, 
1. During the build step instead of generating the DLLs in c:\lib you can generate them in C:\Build\winodbc64\lib  folder
2. The driver and folder name contains 1.0 in its name and this can be confusing with the product version number 

Will open a JIRA for the above 2 changes

**Comment by xwq** [comment]
@hegdean 
1. Now the DLL will be generated in {#BUILDDIR}\lib\x64\Release\ , not hardcode path.
2. I have tried change the path to 1.x , but the driver didn't works. I am not familiar with the registry of the windows platform .  Maybe we can remove the version number on the folder name .

**Comment by robertamarton** [comment]
Is this ready to merge?

**Comment by hegdean** [comment]
Yes Roberta. Go ahead and merge.  Will submit a JIRA for the remaining work

**Pull request merged**